File: Motocitizen/src/motocitizen/Activity/ConfirmDialog.java
package motocitizen.Activity;

import android.app.Activity;
import android.app.AlertDialog;
import android.app.Dialog;
import android.app.DialogFragment;
import android.content.DialogInterface;
import android.os.Bundle;

public class ConfirmDialog extends DialogFragment {

    public static ConfirmDialog newInstance(String title) {

        ConfirmDialog dialogFragment = new ConfirmDialog();
        Bundle bundle = new Bundle();
        bundle.putString("title", title);
        dialogFragment.setArguments(bundle);
        return dialogFragment;
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {

        return new AlertDialog.Builder(getActivity())
                .setTitle(getArguments().getString("title"))
                .setIcon(android.R.drawable.ic_dialog_alert)
                .setPositiveButton(android.R.string.yes,
                        new DialogInterface.OnClickListener() {
                            public void onClick(DialogInterface dialog, int whichButton) {
                                getTargetFragment().onActivityResult(getTargetRequestCode(), Activity.RESULT_OK, getActivity().getIntent());
                            }
                        }
                )
                .setNegativeButton(android.R.string.no, new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int whichButton) {
                        getTargetFragment().onActivityResult(getTargetRequestCode(), Activity.RESULT_CANCELED, getActivity().getIntent());
                    }
                })
                .create();
    }
}

File: Motocitizen/src/motocitizen/ui/dialogs/details/ConfirmDialog.kt
package motocitizen.ui.dialogs.details

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import androidx.fragment.app.DialogFragment

class ConfirmDialog : DialogFragment() {
    companion object {
        private const val YES_STRING = "OK"
        private const val NO_STRING = "Cancel"
        private const val ICON = android.R.drawable.ic_dialog_alert
    }

    var title: String? = ""

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val dialog = AlertDialog.Builder(requireContext())
            .setTitle(title ?: "")
            .setIcon(ICON)
            .setPositiveButton(YES_STRING) { _, _ ->
                parentFragmentManager.setFragmentResult("confirm_dialog_result", Bundle().apply {
                    putBoolean("confirmed", true)
                })
            }
            .setNegativeButton(NO_STRING) { _, _ ->
                parentFragmentManager.setFragmentResult("confirm_dialog_result", Bundle().apply {
                    putBoolean("confirmed", false)
                })
            }
        return dialog.create()
    }
}
